lintcheck/src/json.rs

```diff
@@ -52,14 +57,153 @@ fn load_warnings(path: &Path) -> Vec<LintJson> {
     serde_json::from_slice(&file).unwrap_or_else(|e| panic!("failed to deserialize {}: {e}", path.display()))
 }
 
-fn print_warnings(title: &str, warnings: &[LintJson]) {
+pub(crate) fn diff(old_path: &Path, new_path: &Path, truncate: bool) {
+    let old_warnings = load_warnings(old_path);
+    let new_warnings = load_warnings(new_path);
+
+    let mut added = Vec::new();
+    let mut removed = Vec::new();
+    let mut changed = Vec::new();
+
+    for change in itertools::merge_join_by(old_warnings, new_warnings, |old, new| old.key().cmp(&new.key())) {
+        match change {
+            EitherOrBoth::Both(old, new) => {
+                if old.rendered != new.rendered {
+                    changed.push((old, new));
+                }
+            },
+            EitherOrBoth::Left(old) => removed.push(old),
+            EitherOrBoth::Right(new) => added.push(new),
+        }
+    }
+
+    let lint_warnings = group_by_lint(added, removed, changed);
```

Comment:
If we put the lint name first in the `key` function we could create `lint_warnings` directly via something like

```rust
for (name, chunk) in &merge_join_by(...).chunk_by(...) {}
// or however it needs to work for `ChunkBy`'s lifetime
```

rather than resorting/grouping it

Reply:
I like the idea. I currently can't figure out how to get `chunk_by` working on [`MergeBy`](https://docs.rs/itertools/latest/itertools/structs/struct.MergeBy.html). You suggestion produces the following error:

```
error[E0599]: no method named `chunk_by` found for struct `MergeBy` in the current scope
  --> src/json.rs:69:120
   |
69 |     for (_name, change) in &itertools::merge_join_by(old_warnings, new_warnings, |old, new| old.key().cmp(&new.key())).chunk_by(|change| change.into_left().lint.as_str()) {
   |                                                                                                                        ^^^^^^^^
   |
help: there is a method `chunks` with a similar name
   |
69 |     for (_name, change) in &itertools::merge_join_by(old_warnings, new_warnings, |old, new| old.key().cmp(&new.key())).chunks(|change| change.into_left().lint.as_str()) {
   |            
```

Do you know how to solve it, otherwise I would just manually chunck them.

Reply:
Ah it was renamed in itertools 0.13 and we're on 0.12